src/theory/datatypes/sygus_datatype_utils.cpp

```diff
@@ -540,6 +541,27 @@ TypeNode substituteAndGeneralizeSygusType(TypeNode sdt,
   return sdtS;
 }
 
+TypeNode generalizeSygusType(TypeNode sdt)
+{
+  std::unordered_set<Node> syms;
+  getFreeSymbolsSygusType(sdt, syms);
+  if (syms.empty())
+  {
+    return sdt;
+  }
+  std::vector<Node> svec;
+  std::vector<Node> vars;
+  NodeManager* nm = NodeManager::currentNM();
+  for (const Node& s : syms)
+  {
+    svec.push_back(s);
+    std::stringstream ss;
+    ss << s;
+    vars.push_back(nm->mkBoundVar(ss.str(), s.getType()));
```

Comment:
Wouldn't it be better to get the name directly?
```suggestion
    vars.push_back(nm->mkBoundVar(ss.getName(), s.getType()));
```